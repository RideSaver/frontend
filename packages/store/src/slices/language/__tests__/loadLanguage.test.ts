import { describe, expect, test } from "@jest/globals";
import { configureStore } from "@reduxjs/toolkit";
import i18n, { locales } from "@ridesaver/internationalization";

import type { RootState } from "../../..";

// import { slice, loadLocale } from "..";
const { slice, loadLocale } = require("..") as typeof import("..");

function setupStore() {
    return configureStore({
        reducer: {
            language: slice.reducer,
        },
    });
}

describe("loads language", () => {
    test("load en_PS ", async () => {
        expect.assertions(1);
        const store = setupStore();
        const messages = await store.dispatch(loadLocale("en_PS")).unwrap();
        expect(messages.messages).toMatchSnapshot("en_PS messages");
    });
    test("loads all possible locales correctly", async () => {
        const store = setupStore();
        expect.assertions(locales.length);
        await Promise.all(
            locales.map(async (locale) => {
                await store.dispatch(loadLocale(locale));
                const state = store.getState().language;
                expect(state.messages[locale]).toBeDefined();
                expect(() => i18n._("locale")).not.toThrow();
                expect(i18n.activate).toHaveBeenCalledWith(locale);
                expect(i18n.load).toHaveBeenCalledWith(
                    state.messages[locale].messages
                );
                expect(i18n.loadLocaleData).toHaveBeenCalledWith(
                    state.messages[locale].plurals
                );
            })
        );
    });
});
