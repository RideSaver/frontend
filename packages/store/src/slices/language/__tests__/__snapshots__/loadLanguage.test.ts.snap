// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads language load en_PS : en_PS messages 1`] = `
{
  "messages": {
    "Medium": "Mēďĩũm",
    "Strong": "Śţŕōńĝ",
    "Very Strong": "Vēŕŷ Śţŕōńĝ",
    "Very Weak": "Vēŕŷ Ŵēàķ",
    "Weak": "Ŵēàķ",
    "language": "ĺàńĝũàĝē",
    "{0}": [
      [
        "0",
      ],
    ],
  },
}
`;
